src/pages/RepoPage/FailedTestsTab/FailedTestsPage/SelectorSection/SelectorSection.tsx

```diff
@@ -15,36 +16,51 @@ import Select from 'ui/Select'
 
 import BranchSelector from './BranchSelector'
 
+import { TestResultsFilterParameterType } from '../hooks/useInfiniteTestResults/useInfiniteTestResults'
+
 interface URLParams {
   provider: string
   owner: string
   repo: string
   branch?: string
 }
 
+export const defaultQueryParams = {
+  term: '',
+  flags: [] as string[],
+  historicalTrend: '' as MEASUREMENT_INTERVAL_TYPE,
+  parameter: '' as TestResultsFilterParameterType,
+  testSuites: [] as string[],
+}
+
 const getDecodedBranch = (branch?: string) =>
   !!branch ? decodeURIComponent(branch) : undefined
 
 function SelectorSection() {
-  const { params, updateParams } = useLocationParams({
-    search: '',
-    historicalTrend: '',
-    testSuites: [],
-    flags: [],
-  })
+  const { params, updateParams } = useLocationParams(defaultQueryParams)
 
   // @ts-expect-error need to type out useLocationParams
-  const [selectedFlags, setSelectedFlags] = useState(params?.flags)
-
+  const [selectedFlags, setSelectedFlags] = useState(params?.flags || [])
   const [flagSearch, setFlagSearch] = useState('')
 
   const [selectedTestSuites, setSelectedTestSuites] = useState(
     // @ts-expect-error need to type out useLocationParams
-    params?.testSuites
+    params?.testSuites || []
```

Comment:
do we still need these || [] if the default query param is already []? 

Similar story for flags and the useEffect below

Reply:
We don't need it for useState, but we do need to update params in useEffect. 

When you call updateParams(defaultQueryParams), the params object updates but this might not automatically trigger a state update for selectedTestSuites/Flags because React doesn't directly tie state variables to external objects like params unless you explicitly update the state:


https://github.com/user-attachments/assets/c71c1a75-5fce-448c-92c0-9239cceba27e